shakenfist/image_resolver_debian.py

```diff
@@ -0,0 +1,54 @@
+
+import os
+import requests
+
+from shakenfist import exceptions
+from shakenfist import logutil
+from shakenfist import util
+
+DEBIAN_URL = ('https://cloud.debian.org/images/cloud/OpenStack/current-%(version)s/'
+              'debian-%(version)s-openstack-amd64.qcow2')
+CHECKSUM_URL = 'https://cloud.debian.org/images/cloud/OpenStack/current-%(version)s/MD5SUMS'
+
+
+LOG, _ = logutil.setup(__name__)
+
+
+def resolve(name):
+    # Name is assumed to be in the form debian or debian:10
+    if name.startswith('debian:'):
+        try:
+            _, vernum = name.split(':')
+        except Exception:
+            raise exceptions.VersionSpecificationError(
+                'Cannot parse version: %s' % name)
+
+        url = DEBIAN_URL % {'version': vernum}
+        checksum_url = CHECKSUM_URL % {'version': vernum}
+        resp = requests.head(url,
+                             headers={'User-Agent': util.get_user_agent()})
+        if resp.status_code != 200:
+            raise exceptions.HTTPError(
+                'Failed to fetch %s, status code %d' % (url, resp.status_code))
+
+    else:
+        found_any = False
+        for vernum in range(9, 20):
```

Comment:
Would there be less lines of code if you searched backwards and stopped when a valid URL was found?

Reply:
So, Debian only has 9 and 10 online at the moment. 10 is the current stable. So this is more future proofing than anything else. I guess I could load the index page and parse, but I was being lazy.